api/backend/ecs/task_manager.go

```diff
@@ -50,88 +52,66 @@ func (this *ECSTaskManager) ListTasks() ([]string, error) {
 	return taskARNs, nil
 }
 
-func (this *ECSTaskManager) GetTask(environmentID, taskID string) (*models.Task, error) {
-	ecsEnvironmentID := id.L0EnvironmentID(environmentID).ECSEnvironmentID()
-	ecsTaskID := id.L0TaskID(taskID).ECSTaskID()
+func (this *ECSTaskManager) GetTask(environmentID, taskARN string) (*models.Task, error) {
+	clusterName := id.L0EnvironmentID(environmentID).ECSEnvironmentID()
+	task, err := this.ECS.DescribeTask(clusterName.String(), taskARN)
+	if err != nil {
+		return nil, err
+	}
+
+	return modelFromTasks([]*ecs.Task{task})
+}
 
-	tasks, err := getTaskARNs(this.ECS, ecsEnvironmentID, stringp(ecsTaskID.String()))
+func (this *ECSTaskManager) GetEnvironmentTasks(environmentID string) (map[string]*models.Task, error) {
+	clusterName := id.L0EnvironmentID(environmentID).ECSEnvironmentID()
+	taskDescriptions, err := this.ECS.DescribeEnvironmentTasks(clusterName.String(), id.PREFIX)
 	if err != nil {
 		return nil, err
 	}
 
-	taskDescs := []*ecs.Task{}
-	if len(tasks) > 0 {
-		taskDescs, err = this.describeTasks(ecsEnvironmentID, tasks)
+	taskARNModels := map[string]*models.Task{}
+	for _, taskDescription := range taskDescriptions {
+		task, err := modelFromTasks([]*ecs.Task{taskDescription})
 		if err != nil {
 			return nil, err
 		}
+
+		taskARNModels[aws.StringValue(taskDescription.TaskArn)] = task
 	}
 
-	return modelFromTasks(taskDescs)
+	return taskARNModels, nil
 }
 
-func (this *ECSTaskManager) DeleteTask(environmentID, taskID string) error {
+func (this *ECSTaskManager) DeleteTask(environmentID, taskARN string) error {
 	ecsEnvironmentID := id.L0EnvironmentID(environmentID).ECSEnvironmentID()
-	ecsTaskID := id.L0TaskID(taskID).ECSTaskID()
-
-	taskARNs, err := getTaskARNs(this.ECS, ecsEnvironmentID, stringp(ecsTaskID.String()))
-	if err != nil {
-		return err
-	}
-
-	// This stops the task, later reaping by AWS will prevent it from being returned.
-	reason := "Task stopped by User"
-
-	for _, taskARN := range taskARNs {
-		if err := this.ECS.StopTask(ecsEnvironmentID.String(), reason, *taskARN); err != nil {
-			return err
-		}
-	}
-
-	return nil
+	return this.ECS.StopTask(ecsEnvironmentID.String(), taskARN, StopTaskReason)
```

Comment:
why don't we have to loop through `taskARNs` anymore?

Reply:
tasks only have a single copy